engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportToSizeOf.java

```diff
@@ -80,20 +92,28 @@ public boolean isTheDefaultInstance() {
 
     @Override
     public void update() {
-        if (defaultDynamicFBO == null) {
-            FBO fbo = frameBuffersManager.get(fboName);
-            task.setDimensions(fbo.width(), fbo.height());
-        } else {
-            task.setDimensions(defaultDynamicFBO.width(), defaultDynamicFBO.height());
-        }
+        task.setDimensions(getWidth(), getHeight());
     }
 
     @Override
     public String toString() { // TODO: used for logging purposes at the moment, investigate different methods
         return String.format("%30s: %s", this.getClass().getSimpleName(), fboName);
     }
 
-    public ResourceUrn getFboName() {
-        return fboName;
+    public int getWidth() {
+        FBO fbo = getFbo();
+        return fbo.width();
+    }
+
+    public int getHeight() {
+        FBO fbo = getFbo();
+        return fbo.height();
+    }
+
+    private FBO getFbo() {
+        if (defaultDynamicFBO == null)
+            return frameBuffersManager.get(fboName);
+
+        return defaultDynamicFBO.getFbo();
```

Comment:
I'm not sure whether a ternary statement would make the code more beautiful or any more easier to read, but if this project's code formatting guidelines requires it, I'll do it.